### PR TITLE
SOLR-10654: Override PrometheusResponseWriter Content Type

### PR DESCRIPTION
**Pull request by mlbiscoc**
https://issues.apache.org/jira/browse/SOLR-10654

# Description

The Prometheus Response Writer response content type was JSON which is incorrect.

# Solution

Based on [Prometheus](https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md) docs, override content-type to `text/plain; version=0.0.4`

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by dsmiley** [comment]
Did you at least test manually?  (this is totally adequate)

**Comment by mlbiscoc** [comment]
> Did you at least test manually? (this is totally adequate)

Yes, here is the before:
```
curl -I "localhost:8983/solr/admin/metrics?wt=prometheus"
HTTP/1.1 200 OK
Content-Security-Policy: default-src 'none'; base-uri 'none'; connect-src 'self'; form-action 'self'; font-src 'self'; frame-ancestors 'none'; img-src 'self' data:; media-src 'self'; style-src 'self' 'unsafe-inline'; script-src 'self'; worker-src 'self';
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
Content-Type: application/json;charset=utf-8
Content-Length: 0
```
Here is after:

```
curl -I "localhost:8983/solr/admin/metrics?wt=prometheus"
HTTP/1.1 200 OK
Content-Security-Policy: default-src 'none'; base-uri 'none'; connect-src 'self'; form-action 'self'; font-src 'self'; frame-ancestors 'none'; img-src 'self' data:; media-src 'self'; style-src 'self' 'unsafe-inline'; script-src 'self'; worker-src 'self';
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
Content-Type: text/plain; version=0.0.4
Content-Length: 0
```

File: solr/core/src/java/org/apache/solr/response/PrometheusResponseWriter.java [hunk 1]
```diff
@@ -49,6 +49,7 @@
  */
 @SuppressWarnings(value = "unchecked")
 public class PrometheusResponseWriter extends RawResponseWriter {
+  static String CONTENT_TYPE_PROMETHEUS = "text/plain; version=0.0.4";
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/response/PrometheusResponseWriter.java:52]: obviously should be private and final.

IMO is also a pointless constant but we have different preferences on this

File: solr/core/src/java/org/apache/solr/response/PrometheusResponseWriter.java [hunk 2]
```diff
@@ -49,7 +49,7 @@
  */
 @SuppressWarnings(value = "unchecked")
 public class PrometheusResponseWriter extends RawResponseWriter {
-  static String CONTENT_TYPE_PROMETHEUS = "text/plain; version=0.0.4";
+  private final String CONTENT_TYPE_PROMETHEUS = "text/plain; version=0.0.4";
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/response/PrometheusResponseWriter.java:52]:
```suggestion
  private static final String CONTENT_TYPE_PROMETHEUS = "text/plain; version=0.0.4";
```

**Review by anshumg — APPROVED** [review]

**Pull request merged**